libs/common-services/src/main/java/uk/gov/di/ipv/core/library/service/ConfigService.java

```diff
@@ -44,159 +32,89 @@
-                final Map<String, String> overrides = ssmProvider.getMultiple(basePath);
-                if (overrides.containsKey(terminal)) {
-                    return overrides.get(terminal);
-                } else {
-                    LOGGER.debug(
-                            (new StringMapMessage())
-                                    .with(
-                                            LOG_MESSAGE_DESCRIPTION.getFieldName(),
-                                            "Parameter not present for featureSet")
-                                    .with(LOG_PARAMETER_PATH.getFieldName(), templatePath)
-                                    .with(LOG_FEATURE_SET.getFieldName(), fs));
-                }
+            if (secretValue != null) {
+                Map<String, String> secret =
+                        OBJECT_MAPPER.readValue(secretValue, new TypeReference<>() {});
+                return secret.get(API_KEY);
             }
+            LOGGER.warn(
+                    LogHelper.buildLogMessage("API key not found")
+                            .with(
+                                    LOG_PARAMETER_PATH.getFieldName(),
+                                    formatPath(configurationVariable.getPath(), pathProperties)));
+            return null;
+        } catch (JsonProcessingException e) {
+            LOGGER.error(
+                    LogHelper.buildLogMessage(
+                                    "Failed to parse the api key secret from secrets manager")
+                            .with(
+                                    LOG_PARAMETER_PATH.getFieldName(),
+                                    formatPath(configurationVariable.getPath(), pathProperties)));
+            ;
+            return null;
         }
-        return ssmProvider.get(resolvePath(templatePath, pathProperties));
     }
 
-    private String resolveBasePath() {
-        return String.format(CORE_BASE_PATH, getEnvironmentVariable(ENVIRONMENT));
+    private String formatPath(String path, String... pathProperties) {
+        return String.format(path, (Object[]) pathProperties);
     }
 
-    protected String resolvePath(String path, String... pathProperties) {
-        return resolveBasePath() + String.format(path, (Object[]) pathProperties);
-    }
-
-    private String resolveFeatureSetPath(String featureSet, String path, String... pathProperties) {
-        return resolveBasePath()
-                + String.format("features/%s/", featureSet)
-                + String.format(path, (Object[]) pathProperties);
+    public boolean isLocalDev() {
+        return "true".equals(getEnvironmentVariable(EnvironmentVariable.LOCAL_DEV));
     }
 
+    // PYIC-7048 Replace this with proper config
     public long getBearerAccessTokenTtl() {
         return Optional.ofNullable(getEnvironmentVariable(BEARER_TOKEN_TTL))
                 .map(Long::valueOf)
                 .orElse(DEFAULT_BEARER_TOKEN_TTL_IN_SECS);
     }
```

Comment:
See https://govukverify.atlassian.net/browse/PYIC-7048 - I don't think there's any need for this to be an environment variable